mpmath/libmp/libmpf.py

```diff
@@ -1157,9 +1163,23 @@ def round_digits(digits, dps, base, rounding=round_nearest):
     return digits, exponent
 
 
+def abs_rounding(rounding, sign):
+    """
+    Take into account the sign to determine which kind of rounding the actual
+    digits will use.
+    """
+
+    if rounding == round_ceiling:
+        return round_down if sign else round_up
+    elif rounding == round_floor:
+        return round_up if sign else round_down
+    else:
+        return rounding
+
 
 def to_str(s, dps, strip_zeros=True, min_fixed=None, max_fixed=None,
-           show_zero_exponent=False, base=10, binary_exp=False):
+           show_zero_exponent=False, base=10, binary_exp=False,
+           rounding=round_nearest):
```

Comment:
So, lets not add new rounding modes.

Instead this argument should be set to some sentinel value (None, perhaps) per default.  With this you should emit DeprecationWarning and fallback on legacy code.  Else - use specified rounding mode.

Reply:
So `to_str` should tie to even instead of away from zero?
We could also do something like
```python
round_nearest = sys.intern('n')  # Round to nearest with tying to even
round_nearest_az = sys.intern('m')  # Round to nearest with tying away from zero
```
and set the default value for `rounding` in `to_str` as `round_nearest_az`. This would keep compatibility with previous versions and not require to deprecate the old code.

Reply:
> So to_str should tie to even instead of away from zero?

No.  I think that to_str() should keep current behaviour, but with a deprecation notice.  If user supply a different rounding value (not None) - it should be used.

I don't think that we should add support for a new rounding mode beyond MPFR.

Also, using different defaults in the mp context and for decimal output looks odd.